### ISSUE TITLE
jasmine-client-integration tests keep on loading

### ISSUE THREAD

**Issue by cgudrian** [Sanjo/meteor-jasmine#229]
I cannot get the client integration tests to work: the HTML reporter never leaves the "Loading" state. The application is the pristine "Leaderboard" example and the integration tests have been created by pressing the appropriate button in the reporter window.

I'm using Meteor 1.1.0.2 on Mac OS and Windows 7.

Here's the output of `meteor list`:
```
autopublish             1.0.3  Publish the entire database to all clients
insecure                1.0.3  Allow all database writes by default
meteor-platform         1.2.2  Include a standard set of Meteor packages in your app
sanjo:jasmine           0.14.0  Easily use Jasmine in Meteor
velocity:html-reporter  0.6.2  Reactive Velocity test reports in your app.
```

This is the output of the `meteor` command running in debug mode:
```
[[[[[ ~/devel/webapps/leaderboard ]]]]]       

=> Started proxy.                             
=> Started MongoDB.                           
I20150615-19:27:29.873(2)? ** You've set up some data subscriptions with Meteor.publish(), but
I20150615-19:27:29.874(2)? ** you still have autopublish turned on. Because autopublish is still
I20150615-19:27:29.874(2)? ** on, your Meteor.publish() calls won't have much effect. All data
I20150615-19:27:29.874(2)? ** will still be sent to all clients.
I20150615-19:27:29.874(2)? **
I20150615-19:27:29.874(2)? ** Turn off autopublish by removing the autopublish package:
I20150615-19:27:29.874(2)? **
I20150615-19:27:29.874(2)? **   $ meteor remove autopublish
I20150615-19:27:29.875(2)? **
I20150615-19:27:29.875(2)? ** .. and make sure you have Meteor.publish() and Meteor.subscribe() calls
I20150615-19:27:29.875(2)? ** for each collection that you want clients to see.
I20150615-19:27:29.875(2)? 
I20150615-19:27:29.875(2)? [velocity] adding velocity core
I20150615-19:27:29.875(2)? [velocity] Not adding mirror-registrar because NODE_ENV is development and IS_MIRROR is false
I20150615-19:27:29.926(2)? [velocity] Register framework jasmine-server-integration with regex ^tests/jasmine/server/integration/.+\.(js|coffee|litcoffee|coffee\.md)$
I20150615-19:27:29.927(2)? [velocity] Velocity not started. Queueing func
I20150615-19:27:29.927(2)? [velocity] Register framework jasmine-client-integration with regex ^tests/jasmine/client/integration/.+\.(js|coffee|litcoffee|coffee\.md)$
I20150615-19:27:29.928(2)? [velocity] Velocity not started. Queueing func
I20150615-19:27:29.928(2)? [velocity] Register framework jasmine-client-unit with regex ^tests/jasmine/client/unit/.+\.(js|coffee|litcoffee|coffee\.md)$
I20150615-19:27:29.928(2)? [velocity] Velocity not started. Queueing func
I20150615-19:27:29.928(2)? [velocity] Register framework jasmine-server-unit with regex ^tests/jasmine/server/unit/.+\.(js|coffee|litcoffee|coffee\.md)$
I20150615-19:27:29.929(2)? [velocity] Velocity not started. Queueing func
I20150615-19:27:29.934(2)? [velocity] Server startup
I20150615-19:27:29.935(2)? [velocity] app dir /Users/christian/devel/webapps/leaderboard
I20150615-19:27:29.935(2)? [velocity] config = {
I20150615-19:27:29.935(2)?   "jasmine-server-integration": {
I20150615-19:27:29.935(2)?     "regex": "^tests/jasmine/server/integration/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
I20150615-19:27:29.935(2)?     "name": "jasmine-server-integration",
I20150615-19:27:29.936(2)?     "_regexp": {}
I20150615-19:27:29.936(2)?   },
I20150615-19:27:29.936(2)?   "jasmine-client-integration": {
I20150615-19:27:29.936(2)?     "regex": "^tests/jasmine/client/integration/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
I20150615-19:27:29.936(2)?     "name": "jasmine-client-integration",
I20150615-19:27:29.936(2)?     "_regexp": {}
I20150615-19:27:29.936(2)?   },
I20150615-19:27:29.936(2)?   "jasmine-client-unit": {
I20150615-19:27:29.936(2)?     "regex": "^tests/jasmine/client/unit/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
I20150615-19:27:29.936(2)?     "name": "jasmine-client-unit",
I20150615-19:27:29.936(2)?     "_regexp": {}
I20150615-19:27:29.936(2)?   },
I20150615-19:27:29.937(2)?   "jasmine-server-unit": {
I20150615-19:27:29.937(2)?     "regex": "^tests/jasmine/server/unit/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
I20150615-19:27:29.937(2)?     "name": "jasmine-server-unit",
I20150615-19:27:29.937(2)?     "_regexp": {}
I20150615-19:27:29.937(2)?   }
I20150615-19:27:29.937(2)? }
I20150615-19:27:29.937(2)? [velocity] resetting the world
I20150615-19:27:29.937(2)? [velocity] frameworks with disable auto reset: []
I20150615-19:27:29.945(2)? [velocity] Add paths to watcher [ '/Users/christian/devel/webapps/leaderboard/tests' ]
=> Started your app.

=> App running at: http://localhost:3000/
I20150615-19:27:29.984(2)? [velocity] Search framework for path tests/jasmine/client/integration/sample/src/Player.js
I20150615-19:27:29.984(2)? [velocity] Target framework for tests/jasmine/client/integration/sample/src/Player.js is jasmine-client-integration
I20150615-19:27:29.985(2)? [velocity] Search framework for path tests/jasmine/client/integration/sample/src/Song.js
I20150615-19:27:29.985(2)? [velocity] Target framework for tests/jasmine/client/integration/sample/src/Song.js is jasmine-client-integration
I20150615-19:27:29.987(2)? [velocity] Search framework for path tests/jasmine/client/integration/sample/spec/PlayerSpec.js
I20150615-19:27:29.987(2)? [velocity] Target framework for tests/jasmine/client/integration/sample/spec/PlayerSpec.js is jasmine-client-integration
I20150615-19:27:29.987(2)? [velocity] Search framework for path tests/jasmine/client/integration/sample/spec/SpecMatchers.js
I20150615-19:27:29.987(2)? [velocity] Target framework for tests/jasmine/client/integration/sample/spec/SpecMatchers.js is jasmine-client-integration
I20150615-19:27:29.990(2)? [velocity] File scan complete, now watching /tests
I20150615-19:27:29.990(2)? [velocity] Triggering queued startup functions
I20150615-19:27:29.997(2)? [sanjo:jasmine]: No tests for jasmine-client-unit found. Reporting completed.
I20150615-19:27:30.010(2)? [sanjo:jasmine]: Starting mirror for jasmine-client-integration
I20150615-19:27:30.011(2)? [sanjo:jasmine]: No tests for jasmine-server-integration found. Reporting completed.
I20150615-19:27:30.111(2)? [velocity] 1 mirror(s) requested
I20150615-19:27:30.124(2)? [velocity] Mirror process forked with pid 55886
I20150615-19:27:30.124(2)? [velocity] jasmine-client-integration is starting a mirror at http://localhost:53094/.
I20150615-19:27:30.126(2)? [velocity] You can see the mirror logs at: tail -f /Users/christian/devel/webapps/leaderboard/.meteor/local/log/jasmine-client-integration.log
I20150615-19:27:37.685(2)? [velocity] Mirror registered. Handshaking with mirror...
I20150615-19:27:37.702(2)? [velocity] Connected to mirror, setting state to ready { framework: 'jasmine-client-integration',
I20150615-19:27:37.702(2)?   port: '53094',
I20150615-19:27:37.702(2)?   host: 'http://localhost:53094/' }
```

And these are the contents of the log file `jasmine-client-integration.log`:
```
Spawn script arguments:
parentPid: 55866
command: meteor
commandArguments: [ 'run',
  '--test-app',
  '--port',
  '53094',
  '--include-tests',
  'jasmine/client/integration',
  '--settings',
  '/var/folders/0g/w566yxwx59l19c025jnjxsj00000gn/T/tmp-55885cj46wRMPMTsW.tmp',
  '--release',
  'velocity:METEOR@1.1.0.2_3' ]
pid: 55886
child.pid: 55887
=> Using velocity:METEOR@1.1.0.2_3 as requested (overriding Meteor 1.1.0.2)

[[[[[ ~/devel/webapps/leaderboard ]]]]]

=> Started proxy.
I20150615-19:27:37.592(2)? [sanjo:long-running-child-process]: Exposing imports for testing purposes
I20150615-19:27:37.593(2)? ** You've set up some data subscriptions with Meteor.publish(), but
I20150615-19:27:37.593(2)? ** you still have autopublish turned on. Because autopublish is still
I20150615-19:27:37.593(2)? ** on, your Meteor.publish() calls won't have much effect. All data
I20150615-19:27:37.593(2)? ** will still be sent to all clients.
I20150615-19:27:37.593(2)? **
I20150615-19:27:37.593(2)? ** Turn off autopublish by removing the autopublish package:
I20150615-19:27:37.594(2)? **
I20150615-19:27:37.594(2)? **   $ meteor remove autopublish
I20150615-19:27:37.594(2)? **
I20150615-19:27:37.594(2)? ** .. and make sure you have Meteor.publish() and Meteor.subscribe() calls
I20150615-19:27:37.594(2)? ** for each collection that you want clients to see.
I20150615-19:27:37.594(2)? 
I20150615-19:27:37.594(2)? [velocity] Mirror detected - Not adding velocity core
I20150615-19:27:37.594(2)? [velocity] Not adding mirror-registry because NODE_ENV is development  and IS_MIRROR is true
I20150615-19:27:37.594(2)? [velocity] Starting mirror on port 53094
=> Started your app.

=> App running at: http://localhost:53094/
I20150615-19:27:37.665(2)? [velocity] Mirror started. Connecting via DDP to parent
I20150615-19:27:37.680(2)? [velocity] Mirror connected to parent. Registering mirror...
I20150615-19:27:37.706(2)? [velocity] Established connection with Velocity.
```

**Comment by Sanjo** [comment]
Can you upload the leaderboard app please. Thanks.

**Comment by cgudrian** [comment]
https://dl.dropboxusercontent.com/u/9848985/leaderboard.tbz

**Comment by cgudrian** [comment]
Content Security Policy! This fixes this issue:

https://github.com/mad-eye/meteor-mocha-web/issues/119#issuecomment-68628895

**Comment by cgudrian** [comment]
Unfortunately I have to reopen this issue. While the tests work on Windows they still don't on Mac OS. I've already debugged the server code and found out, that the `PARENT_URL` environment option is `undefined` – possibly because the mirror flag is `false` – which eventually leads to errors in the server integration code.

**Comment by cjsauer** [comment]
Yep I also have this issue. Client integration tests get stuck at loading. The command `meteor --test --release velocity:METEOR@1.1.0.2_3` runs without a hitch though.

**Comment by Sanjo** [comment]
I will have a look at the Leaderboard app tomorrow.

**Comment by Sanjo** [comment]
> Unfortunately I have to reopen this issue. While the tests work on Windows they still don't on Mac OS. I've already debugged the server code and found out, that the PARENT_URL environment option is undefined – possibly because the mirror flag is false – which eventually leads to errors in the server integration code.

@cgudrian The leaderboard app that you have uploaded has only server integration tests. Are you now talking about a new issue?

**Comment by Sanjo** [comment]
I cannot reproduce the "jasmine-client-integration tests keep on loading" issue. I start with the leaderboard app without tests directory. Then I click "Add jasmine-client-integration sample tests". Then it takes 12 seconds loading and then I get the results. Maybe it just takes longer to load then you would expect?

**Comment by cgudrian** [comment]
The original problem has been solved. The browser-policy package (which comes bundled with meteor-boilerplate) was responsible for the jasmine frame not being loaded. After correctly configuring the package (as mentioned above) the client unit tests show up.

As for the `PARENT_URL` issue: it only shows up sporadically. Further investigation on my side is needed. I'll open another issue if necessary.

Thanks for your work and sorry for the confusion!

**Comment by cgudrian** [comment]
I've finally found out why the client integration tests weren't loading: somehow the flag "shouldRunClientIntegrationTests" was set to ``true`` in the local storage of the main site (http://localhost:3000). After executing
```
localStorage.removeItem("shouldRunClientIntegrationTests")
```
in the browser's JavaScript console everything works as expected.

**Comment by Sanjo** [comment]
Ok. This should only happen when the URL has `?jasmine=true`. I will add another check.

**Comment by DerrickJames** [comment]
I'm also experiencing the same issue, my app runs as a cluster using meteorhacks:clusters. When I disconnect all my connections to my services, my client integration tests work perfectly but when I reconnect the services they get stuck at loading. Not sure whether this package is not happy working with micro-services? Here's a [similar raised issue](https://github.com/meteorhacks/cluster/issues/47) I found but I guess it was posted at the wrong place, I also believe he's also shared some sample code here [Cluster with velocity](https://github.com/boldkhuu/cluster-with-velocity)

**Comment by Sanjo** [comment]
Duplicate of https://github.com/Sanjo/meteor-jasmine/issues/228. Fixed with 0.15.3
